src/dialog/tooltip/tooltip.directive.ts

```diff
@@ -73,6 +80,20 @@ export class TooltipDirective extends DialogDirective {
 		this.dialogConfig.compID = "tooltip-" + TooltipDirective.tooltipCounter;
 		this.dialogConfig.content = this.ibmTooltip;
 		this.dialogConfig.type = this.tooltipType;
-		this.descriptorId = this.dialogConfig.compID;
+	}
+
+	close() {
+		this.dialogService.close(this.viewContainerRef, false);
+		this.expanded = false;
+	}
+
+	@HostListener("keydown", ["$event"])
+	onKeydown(event: KeyboardEvent) {
```

Comment:
This is already handled in `dialog.directive`

Reply:
`dialog.directive` only handles click events

Reply:
Which also handles enter and space (for buttons) ... if we want to make sure we handle keydowns, add the appropriate bindings in `dialog.directive`

Reply:
So I have to add keydowns because enter/space was not opening the tooltip.. So ill just add it to `dialog.directive`